src/main/java/com/thoughtworks/gauge/datastore/DataStoreFactory.java

```diff
@@ -5,6 +5,14 @@
  *----------------------------------------------------------------*/
 package com.thoughtworks.gauge.datastore;
 
+import static com.thoughtworks.gauge.GaugeConstant.ENABLE_MULTITHREADING_ENV;
+
+
+/**
+ * @deprecated DataStoreFactory is no longer valid. In multithreading execution it strictly throw Exception.
```

Comment:
"In multithreading execution it strictly throw Exception." -> "In multithreading execution an Exception will be thrown prompting this change."